tests/v2/validation/charts/monitoring.go

```diff
@@ -0,0 +1,430 @@
+	if len(activeTargets) < 1 {
+		return false, fmt.Errorf("Failed to find any active targets")
+	}
+
+	for _, target := range activeTargets {
+		targetMap := target.(map[string]interface{})
+		if targetMap["health"].(string) == "down" {
+			downTargets = append(downTargets, targetMap["labels"].(map[string]interface{})["instance"].(string))
+		}
+	}
+	statusInit = len(downTargets) == 0
+
+	if !statusInit {
+		return statusInit, fmt.Errorf("All active target(s) are not healthy: %v", downTargets)
+	}
+
+	return statusInit, nil
+}
+
+// editAlertReceiver is a private helper function
+// that edits alert config structure to be used by the webhook receiver.
+func editAlertReceiver(alertConfigByte []byte, origin string, originURL *url.URL) (string, error) {
+	alertConfig := config.Config{}
+
+	err := yaml.Unmarshal(alertConfigByte, &alertConfig)
+	if err != nil {
+		return "", fmt.Errorf("Failed to unmarshal alert config: %v", err)
+	}
+
+	alertConfig.Global = &config.GlobalConfig{
+		ResolveTimeout: alertConfig.Global.ResolveTimeout,
+	}
+	alertConfig.Receivers = append(alertConfig.Receivers, &config.Receiver{
+		Name: webhookReceiverDeploymentName,
+		WebhookConfigs: []*config.WebhookConfig{
+			{
+				HTTPConfig: &config.HTTPClientConfig{
+					ProxyURL: config.URL{URL: originURL},
+				},
+				NotifierConfig: config.NotifierConfig{
+					VSendResolved: false,
+				},
+				URL: origin,
+			},
+		},
+	})
+
+	stringifiedAlertConfig := alertConfig.String()
+	encodedStringifiedAlertConfig := base64.StdEncoding.EncodeToString([]byte(stringifiedAlertConfig))
+
+	return encodedStringifiedAlertConfig, nil
+}
+
+// editAlertRoute is a private helper function
+// that edits alert config structure to be used by the webhook receiver.
+func editAlertRoute(alertConfigByte []byte, origin string, originURL *url.URL) (string, error) {
+	alertConfig := config.Config{}
+
+	err := yaml.Unmarshal(alertConfigByte, &alertConfig)
+	if err != nil {
+		return "", fmt.Errorf("Failed to unmarshal alert config: %v", err)
+	}
+
+	alertConfig.Global = &config.GlobalConfig{
+		ResolveTimeout: alertConfig.Global.ResolveTimeout,
+	}
+	alertConfig.Route.Routes = append(alertConfig.Route.Routes, &config.Route{
+		GroupWait:      alertConfig.Route.GroupWait,
+		GroupInterval:  alertConfig.Route.GroupInterval,
+		RepeatInterval: alertConfig.Route.RepeatInterval,
+		Match:          ruleLabel,
+		Receiver:       webhookReceiverDeploymentName,
+	})
+
+	stringifiedAlertConfig := alertConfig.String()
+	encodedStringifiedAlertConfig := base64.StdEncoding.EncodeToString([]byte(stringifiedAlertConfig))
+
+	return encodedStringifiedAlertConfig, nil
+}
+
+// createPrometheusRule is a private helper function
+// that creates a prometheus rule to be used by the webhook receiver.
+func createPrometheusRule(client *rancher.Client, clusterID string) error {
+	ruleName := "webhook-rule-" + namegenerator.RandStringLower(defaultRandStringLength)
```

Comment:
I honestly think this should be an extension, this seems like something useful others could use